### PR TITLE
feat: map screen improv

### PR DESCRIPTION
**Pull request by Rania5724**
## Description

This PR focuses on making the map feel smoother and more consistent while also cleaning up code quality issues.

#### **1. SonarCloud cleanup**
I cleaned up the code smells that were safe to remove. These were small things like redundant checks, unused branches, code complexity and minor structural issues. The behavior stays the same, but the codebase is a bit more lightweight and easier to read.

#### **2. Replaced instant camera jumps with `flyTo`**
Anywhere the map used to snap instantly to a new position, it now uses `flyTo`. This applies when opening the map, when recentering, and when selecting a pin. The result is much smoother and feels more intentional.

#### **3. Fixed the selected-pin state bug**
Before this fix, selecting a pin made it grow and start its ripple animation, but if you opened the post and came back, the pin stayed big and the ripple was gone. It also became impossible to shrink it again. The state now resets correctly whenever you leave the map, so pins behave normally again when you return.

#### **4. Automatic zoom on the first available post**
The map no longer defaults to Lausanne when opening. It now checks the current tab, finds the first post, and flies to it right away. This avoids the awkward empty default view and makes the screen feel more responsive and useful.

#### **5. Zoom-based pin clustering**
I introduced a small clustering layer on top of the existing map pins:

- At high zoom (`zoomBand >= 11`), there is no clustering and all pins are shown individually.
- At medium zoom (`zoomBand 9–10`), nearby pins within a 2km radius are merged into small clusters.
- At low zoom (`zoomBand <= 8`), pins within a larger 8km radius are merged into bigger clusters.

The clustering itself is done with a simple union–find structure: any pins that are within the configured radius are grouped together and turned into a single `ClusterPin`. The cluster position is computed as the average latitude/longitude of all pins in the group, and the `count` field reflects how many items were merged.  

I also added a unit test that covers the clustering behavior to make sure the distance-based grouping and zoom thresholds behave as expected.

#### **6. Map and profile map follow the app theme**
When the user changes the theme in the app, the Mapbox style updates accordingly, keeping the whole UI visually consistent instead of only following the phone system theme.

## UI Preview
<table>
<tr>
<td></td>
<td></td>
<td></td>
</tr>
</table>

## Related issues
This PR closes issue #278.

## Trade-offs / known limitations
None found so far. All changes keep the existing logic intact while cleaning up the code. The clustering distances aren’t also final. If the clustering feels too strong or too weak, we can always adjust the radius numbers later.

## Additional Remarks
- To reduce Map Pins complexity, I split some logic into smaller internal methods and added more targeted tests.
- Like I mentioned in the previous map PR's,  pushing the test coverage further would start requiring heavier setup, so I kept the extra tests focused on the most critical behaviors (pin rendering, image loading and clustering) and I doubt I will be able to make it to 80% line coverage.
- **Disclaimer**: Since I added new elements to MapPins, I used **ChatGPT** to generate the new cluster bitmap as I am not very familiar with this part of the code. I also used it to cover some edge cases I had ignored and so improve the cluster logic, generate some methods documentation and also cover some lines I was unable to test.

### CODE REVIEW

File: app/src/main/java/com/android/wildex/model/map/ClusterPins.kt [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  val groups = buildClusterGroups(basePins, parent)
+
+  // build ClusterPins
+  return buildClusterPins(groups)
+}
+
+/** Handle trivial cases for clustering: empty list or single pin. */
+private fun handleTrivialClusterCases(basePins: List<MapPin>): List<MapPin>? {
+  if (basePins.isEmpty()) return emptyList()
+  if (basePins.size == 1) {
```
Comment (zlissaneeddine) [inline app/src/main/java/com/android/wildex/model/map/ClusterPins.kt]:
```suggestion
  else {
```

File: app/src/main/java/com/android/wildex/model/map/ClusterPins.kt [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    var cur = x
+    while (parent[cur] != cur) {
+      val p = parent[cur]
+      parent[cur] = r
+      cur = p
+    }
+    return r
+  }
+  return basePins.indices
+      .groupBy { i -> findRoot(i) }
```
Comment (zlissaneeddine) [inline app/src/main/java/com/android/wildex/model/map/ClusterPins.kt]: As always, a little nitpicking is always necessary :)
```suggestion
      .groupBy { findRoot(it) }
```

File: app/src/main/java/com/android/wildex/model/map/ClusterPins.kt [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+): List<MapPin> =
+    groups.map { (root, group) ->
+      val avgLat = group.map { it.location.latitude }.average()
+      val avgLon = group.map { it.location.longitude }.average()
+      val count = group.size
+      val clusterId: Id = "cluster_$root"
+
+      MapPin.ClusterPin(
+          id = clusterId,
+          location = Location(avgLat, avgLon, "Cluster"),
```
Comment (zlissaneeddine) [inline app/src/main/java/com/android/wildex/model/map/ClusterPins.kt]: You should define `"Cluster"` in `strings.xml`

File: app/src/main/java/com/android/wildex/ui/map/MapScreenViewModel.kt [hunk 4]
```diff
@@ -181,6 +198,9 @@ class MapScreenViewModel(
   fun onPinSelected(pinId: Id) {
     viewModelScope.launch {
       val pin = _uiState.value.pins.firstOrNull { it.id == pinId } ?: return@launch
+      val newCenter = pin.location
```
Comment (zlissaneeddine) [inline app/src/main/java/com/android/wildex/ui/map/MapScreenViewModel.kt]: You don't really need to create this variable here. You only use it once afterwards.
```suggestion
```

File: app/src/main/java/com/android/wildex/ui/map/MapScreenViewModel.kt [hunk 5]
```diff
@@ -181,6 +198,9 @@ class MapScreenViewModel(
...
+      _uiState.value = _uiState.value.copy(centerCoordinates = newCenter)
```
Comment (zlissaneeddine) [inline app/src/main/java/com/android/wildex/ui/map/MapScreenViewModel.kt]:
```suggestion
      _uiState.value = _uiState.value.copy(centerCoordinates = pin.location)
```

File: app/src/main/java/com/android/wildex/ui/map/MapPins.kt [hunk 6]
```diff
@@ -88,16 +98,7 @@ internal data class BaseKey(
...
- * Composable overlay that displays map pins on a Mapbox MapView.
- *
- * @param modifier Modifier to be applied to the Box containing the map pins.
- * @param mapView The Mapbox MapView where the pins will be displayed.
- * @param pins List of MapPin objects representing the pins to be displayed.
- * @param currentTab The current MapTab, used to determine color scheme.
- * @param selectedId The ID of the currently selected pin, if any.
- * @param onPinClick Callback function to be invoked when a pin is clicked, with the
- */
+/** Composable overlay that displays map pins on a Mapbox MapView. */
```
Comment (zlissaneeddine) [inline app/src/main/java/com/android/wildex/ui/map/MapPins.kt]: Why did you remove the documentation here ? :crying_cat_face:

File: app/src/main/java/com/android/wildex/ui/map/MapPins.kt [hunk 7]
```diff
@@ -115,266 +116,425 @@ fun PinsOverlay(
...
+  }
+}
 
-    for (pin in pins) {
-      val id = pin.id
-      val isSelected = id == selectedId
-      val showExcl = pin is MapPin.ReportPin && pin.assigneeId.isNullOrBlank() && !isSelected
-      val scale = if (isSelected) 1.25f else 1.0f
-      val targetUrl = pin.imageURL.ifBlank { fallbackUrl }
+/** Dispose of PointAnnotationManager and clear caches */
```
Comment (zlissaneeddine) [inline app/src/main/java/com/android/wildex/ui/map/MapPins.kt]: You should put a more detailed documentation here too.
